src/main/groovy/com/epam/petclinic/plugin/CodeCoveragePlugin.groovy

```diff
@@ -0,0 +1,175 @@
+package com.epam.petclinic.plugin
+
+import com.epam.petclinic.plugin.util.CoverageReportUtil
+import org.gradle.api.GradleException
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.plugins.JavaPlugin
+
+/**
+ * Code Coverage Report plugin.
+ *
+ * Date: 5/10/2017
+ *
+ * @author Stanislau Halauniou
+ */
+class CodeCoveragePlugin implements Plugin<Project> {
+
+    private static final String CODE_COVERAGE_TASK_GROUP = 'Petclinic Code Coverage'
+    private static final String JACOCO_PLUGIN_ID = 'jacoco'
+    private static final String XML_REPORT_PATH = '/reports/coverage/coverage-report.xml'
+
+    private Map props
+    private XmlParser parser
+
+    @Override
+    void apply(Project project) {
+        parser = configureParser()
+        props = getMetricProperties()
+
+        project.plugins.withType(JavaPlugin) {
+            addCheckMinimumCoverageTask(project)
+            addCreateJacocoReportTask(project)
+            configureJacoco(project)
+        }
+    }
+
+    /**
+     * Performs the configuration of JaCoCo plugin.
+     *
+     * @param project
+     */
+    private void configureJacoco(Project project) {
+        project.plugins.apply(JACOCO_PLUGIN_ID)
+
+        //JaCoCo version
+        project.jacoco {
+            toolVersion = "${project.jacocoVersion}"
+        }
+
+        //configuration of the JaCoCo specific properties of the test task
+        project.test {
+            jacoco {
+                append = false
+                destinationFile = project.file("${project.buildDir}/jacoco/jacocoTest.exec")
```

Comment:
Define a constant to avoid collisions between data in CoverageReportUtil and here